File: 3-semestre/estrutura de dados/Aula 6/Principal.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Principal {
    
    public static void main(String[] args) {
        List<Aluno> alunos = new ArrayList<>(); 

        Aluno tmp;
        String nome;
        int matricula = 1;
        String opcao;
        Scanner teclado = new Scanner(System.in);

        while(true){
            System.out.print("Digite o nome: ");
            nome = teclado.nextLine().toUpperCase();
            tmp = new Aluno(matricula, nome);

            if(alunos.contains(tmp)){
                System.out.println("Aluno com email ja registrado");
                System.gc();
            } else {
                System.out.println("Aluno cadastrado com a matricula de número "+matricula);
                System.out.println("e com email "+tmp.email);
                alunos.add(tmp);
                matricula++;
            }

            System.out.print("Continuar: [1 - Sim; Outra tecla - Não] ");
            opcao = teclado.nextLine();
            if(!opcao.equals("1")){
                break;
            }
        }

        alunos.sort((aluno1, aluno2) -> aluno1.nome.compareTo(aluno2.nome));

        System.out.println("Relacao alunos");
        for(Aluno aluno : alunos){
            System.out.println(aluno);
        }

        teclado.close();
    }
}

File: 3-semestre/estrutura de dados/Aula 6/principal.py
from aluno import Aluno

class Principal:
    alunos = []
    matricula = 1
    while(True):
        
        nome = input("Digite o nome: ").upper()
        tmp = Aluno(matricula, nome)

        if(tmp in alunos):
            print("Aluno com email ja registrado")
        else: 
            alunos.append(tmp)

            print("Aluno cadastrado com a matricula de número ", matricula)
            print("e com email ", tmp.email)
            matricula+=1
            
        opcao = input("Continuar: [1 - Sim; Outra tecla - Não] ");
        if(opcao != "1"):
            break

    alunos.sort(key=lambda aluno : aluno.nome)

    print("Relacao alunos")
    for aluno in alunos: #for objeto
        print(aluno)

